src/Components/Blueprints/ImportBlueprintModal.tsx

```diff
@@ -50,52 +53,63 @@ export const ImportBlueprintModal: React.FunctionComponent<
     _event: React.ChangeEvent<HTMLInputElement> | React.DragEvent<HTMLElement>,
     file: File
   ) => {
+    setFileContent('');
     setFilename(file.name);
     setIsRejected(false);
     setIsInvalidFormat(false);
   };
+  React.useEffect(() => {
+    if (filename && fileContent) {
+      try {
+        const isToml = filename.endsWith('.toml');
+        const isJson = filename.endsWith('.json');
+        if (isToml) {
+          const tomlBlueprint = parse(fileContent);
+          const blueprintFromFile = mapOnPremToHosted(tomlBlueprint);
+          const importBlueprintState = mapExportRequestToState(
+            blueprintFromFile,
+            []
+          );
+          setImportedBlueprint(importBlueprintState);
+        } else if (isJson) {
+          const blueprintFromFile = JSON.parse(fileContent);
+          const blueprintExportedResponse: BlueprintExportResponse = {
+            name: blueprintFromFile.name,
+            description: blueprintFromFile.description,
+            distribution: blueprintFromFile.distribution,
+            customizations: blueprintFromFile.customizations,
+            metadata: blueprintFromFile.metadata,
+          };
+          const importBlueprintState = mapExportRequestToState(
+            blueprintExportedResponse,
+            blueprintFromFile.image_requests || []
+          );
+          setImportedBlueprint(importBlueprintState);
+        }
+      } catch (error) {
+        setIsInvalidFormat(true);
+        dispatch(
+          addNotification({
+            variant: 'warning',
+            title: 'No blueprint was build',
```

Comment:
I know this is just a copy, but the warning is probably incorrect, we should say something like "File is not a valid blueprint"

Reply:
Mocks are not always 100% tought through, this case I feel like it could have been copy pasted from other mock.
What I'm interested in is whether my comment makes sense to you, or if I just don't get why this might be actually correct message :)

Reply:
Well, I feel like the notification might make sense. We already inform in the warning at the bottom of the text field that the format is not compatible with our blueprints. So this one just made sense to me, so that users know that we did not create any blueprint just from the fact that they uploaded the file. We can discuss this with Kelsea, but I think this is a minor thing:)

Reply:
Yeah, totaly minor, if you feel this is good, lets keep it :)